src/modules/function/declaration.rs

```diff
@@ -98,24 +101,42 @@ impl SyntaxModule<ParserMetadata> for FunctionDeclaration {
                     break
                 }
                 let is_ref = token(meta, "ref").is_ok();
+                let name_token = meta.get_current_token();//saving for possible error
                 let name = variable(meta, variable_name_extensions())?;
                 // Optionally parse the argument type
+                let mut arg_type = Type::Generic;
                 match token(meta, ":") {
                     Ok(_) => {
                         self.arg_refs.push(is_ref);
                         self.arg_names.push(name.clone());
-                        self.arg_types.push(parse_type(meta)?);
+                        arg_type = parse_type(meta)?;
+                        self.arg_types.push(arg_type.clone());
                     },
                     Err(_) => {
                         self.arg_refs.push(is_ref);
                         self.arg_names.push(name.clone());
                         self.arg_types.push(Type::Generic);
                     }
                 }
-                let tok = meta.get_current_token();
-                if token(meta, "=").is_ok() {
-                    return error!(meta, tok, "Default values for function arguments are not yet supported")
+                match token(meta,"="){
+                    Ok(_) => {
+                        optional = true;
+                        let mut expr = Expr::new();
+                        syntax(meta, &mut expr)?;
+                        if arg_type != Type::Generic {
+                            if arg_type != expr.get_type() {
+                                return error!(meta, name_token, "optional arg doesn't match annotated type");
```

Comment:
This looks more formal
```suggestion
                                return error!(meta, name_token, "Optional argument does not match annotated type");
```